AppTaskQueue/appscale/taskqueue/push_worker.py

```diff
@@ -92,132 +96,143 @@ def execute_task(task, headers, args):
+                 url=url, te=str(time_elapsed)))
+        return response.status
+      elif response.status == 302:
+        redirect_url = response.getheader('Location')
+        logger.info(
+          "Task %s asked us to redirect to %s, so retrying there." % (
+            args['task_name'], redirect_url))
+        url = urlparse(redirect_url)
+        if redirects_left == 0:
+          raise task.retry(countdown=wait_time)
+        redirects_left -= 1
       else:
-        connection.putheader('content-type',
-                             'application/x-www-form-urlencoded')
-
-    connection.putheader("Content-Length", str(content_length))
-
-    retries = int(task.request.retries) + 1
-    wait_time = get_wait_time(retries, args)
-
-    try:
-      connection.endheaders()
-      if args["body"]:
-        connection.send(args['body'])
-
-      response = connection.getresponse()
-      response.read()
-      response.close()
-    except (BadStatusLine, SocketError):
-      logger.warning(
-        '{task} failed before receiving response. It will retry in {wait} '
-        'seconds.'.format(task=args['task_name'], wait=wait_time))
-      raise task.retry(countdown=wait_time)
-
-    if 200 <= response.status < 300:
-      # Task successful.
-      item = TaskName.get_by_key_name(args['task_name'])
-      if item is not None:
-        item.state = TASK_STATES.SUCCESS
-        item.endtime = datetime.datetime.now()
-        db.put(item)
-
-      time_elapsed = datetime.datetime.utcnow() - start_time
-      logger.info(
-        '{task} received status {status} from {url} [time elapsed: {te}]'. \
-        format(task=args['task_name'], status=response.status,
-               url=url, te=str(time_elapsed)))
-      return response.status
-    elif response.status == 302:
-      redirect_url = response.getheader('Location')
-      logger.info(
-        "Task %s asked us to redirect to %s, so retrying there." % (
-          args['task_name'], redirect_url))
-      url = urlparse(redirect_url)
-      if redirects_left == 0:
+        message = ('Received a {status} for {task}. '
+                   'Retrying in {wait} secs.'.format(status=response.status,
+                                                     task=args['task_name'],
+                                                     wait=wait_time))
+        logger.warning(message)
         raise task.retry(countdown=wait_time)
-      redirects_left -= 1
-    else:
-      message = ('Received a {status} for {task}. '
-                 'Retrying in {wait} secs.'.format(status=response.status,
-                                                   task=args['task_name'],
-                                                   wait=wait_time))
-      logger.warning(message)
-      raise task.retry(countdown=wait_time)
+  except EventletTimeout as thrown_timeout:
+    if thrown_timeout != timeout:
+      raise
+
+    logger.exception('Task timed out')
```

Comment:
Are there any details for the task that should be logged here?

Reply:
Good point. I added the task ID to the log output.